internal/armada/server/submit_to_log.go

```diff
@@ -361,6 +362,68 @@ func (srv *PulsarSubmitServer) CancelJobs(ctx context.Context, req *api.JobCance
 	}, nil
 }
 
+func (srv *PulsarSubmitServer) CancelJobSet(ctx context.Context, req *api.JobSetCancelRequest) (*types.Empty, error) {
+	if req.Queue == "" {
+		return nil, &armadaerrors.ErrInvalidArgument{
+			Name:    "Queue",
+			Value:   req.Queue,
+			Message: "queue is empty",
+		}
+	}
+	if req.JobSetId == "" {
+		return nil, &armadaerrors.ErrInvalidArgument{
+			Name:    "JobSetId",
+			Value:   req.JobSetId,
+			Message: "JobSetId is empty",
+		}
+	}
+
+	err := validation.ValidateJobSetFilter(req.Filter)
+	if err != nil {
+		return nil, err
+	}
+
+	userId, groups, err := srv.Authorize(ctx, req.Queue, permissions.CancelAnyJobs, queue.PermissionVerbCancel)
```

Comment:
this is a nit but I'd always make the authorize statement the very first thing in the function (I understand this means making authorize reject empty queue/jobset).  I'd do this because otherwise there's a risk that someone adds some code above the authorize that needs the user to be verified before exceution.

This is probably out of scope of the PR so shouldn't block merging, but if you agree we can raise a ticket to audit the existing code to make it so.

Reply:
I agree its probably best practice. We should raise a ticket to go over all uses and make them consistent (auth first)